arch/arm/boot/dts/armada-385-ts7840.dts

```diff
@@ -0,0 +1,840 @@
+				pagesize = <64>;
+			};
+		};
+
+		opencores_spi1: spi@420 {
+			compatible = "opencores,spi-oc";
+			reg = <0 0x420 32>;
+			interrupt-parent = <&fpgairqc>;
+			interrupts = <12 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&fpga_pll0c1>;
+			clock-names ="spi-oc-clk";
+			opencores-spi,idx = <0>;
+			opencores-spi,num-chipselects = <1>;
+
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			spidev@0 {
+				compatible = "spidev";
+				reg = <0>;
+				spi-max-frequency = <1000000>;
+			};
+		};
+
+		mikro_i2c: i2c@500 {
+			compatible = "opencores,i2c-ocores";
+			reg = <0 0x500 4>;
+			interrupt-parent = <&fpgairqc>;
+			interrupts = <13 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&fpga_pll0c1>;
+			clock-names = "i2c-oc-clk";
+		};
+
+		mikro_pwm: pwm@600 {
+			compatible = "technologic,pwm";
+			reg = <0 0x600 8>;
+			clocks = <&fpga_pll0c1>;
+			clock-names = "pwm-input-clk";
+		};
+
+		ts_mmc: ts_mmc@800 {
+			compatible = "ts,tssdcore2";
+			reg = <0 0x800 32>;
+			//interrupt-parent = <&fpgairqc>;
+			//interrupts = <14 IRQ_TYPE_LEVEL_HIGH>;
+			busy-gpios = <&fpga_bank0 15 0>;
+			debug-gpios = <&fpga_bank1 12 0>;
+			status = "disabled";
+		};
+
+		mikro_adc0: adc@900 {
+			compatible = "technologic,ts-simple-adc";
+			reg = <0 0x900 4>;
+			interrupt-parent = <&fpgairqc>;
+			interrupts = <21 IRQ_TYPE_LEVEL_HIGH>;
+		};
+	};
+
+	sfp0: sfp {
+		compatible = "sff,sfp";
+		i2c-bus = <&sfp0_i2c>;
+	};
+
+	sfp1: sfp {
+		compatible = "sff,sfp";
+		i2c-bus = <&sfp1_i2c>;
+	};
+};
+
+&ahci0 {
+	status = "okay";
+};
+
+&bm {
+	status = "okay";
+};
+
+&bm_bppi {
+	status = "okay";
+};
+
+/* magjack/sfp port 0 */
+&eth0 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&ge0_rgmii_pins>;
+	status = "okay";
+	phy = <&phy0>;
+	phy-mode = "rgmii-id";
+	buffer-manager = <&bm>;
+	/*sfp = <&sfp0>;*/
+	bm,pool-long = <0>;
+	bm,pool-short = <1>;
+};
+
+/* magjack/sfgp port 1 */
+&eth1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&ge1_rgmii_pins>;
+	status = "okay";
+	phy = <&phy1>;
+	phy-mode = "rgmii-id";
+	/*sfp = <&sfp1>;*/
+	bm,pool-long = <2>;
+	bm,pool-short = <1>;
+	buffer-manager = <&bm>;
+};
+
+/* sgmii port to eth switch */
+&eth2 {
+	status = "okay";
+	phy-mode = "sgmii";
+	bm,pool-long = <3>;
+	bm,pool-short = <1>;
+	buffer-manager = <&bm>;
+
+	fixed-link {
+		speed = <1000>;
+		full-duplex;
+	};
+};
+
+&gpio0 {
+	gpio-line-names =
+		"SILAB_TXD",
+		"UART0_TXD",
+		"I2C_0_CLK",
+		"I2C_0_DAT",
+		"MPP4_GE_MDC",
+		"MPP5_GE_MDIO",
+		"MPP6_GE_TXCLK",
+		"MPP7_GE_TXD0",
+		"MPP8_GE_TXD1",
+		"MPP9_GE_TXD2",
+		"MPP10_GE_TXD3",
+		"MPP11_GE_TXCTL",
+		"MPP12_GE_RXD0",
+		"MPP13_GE_RXD1",
+		"MPP14_GE_RXD2",
+		"MPP15_GE_RXD3",
+		"MPP16_GE_RXCTL",
+		"MPP17_GE_RXCLK",
+		"WIFI_IRQ_N",
+		"FPGA_RESET",
+		"",
+		"MPP_21_GE_RXD0",
+		"SPI_0_MOSI",
+		"SPI_0_CLK",
+		"SPI_0_MISO",
+		"SPI_0_BOOT_CS0#",
+		"",
+		"MPP_27_GE_TXCLK",
+		"MPP_28_GE_TXD0",
+		"MPP_29_GE_TXD1",
+		"MPP_30_GE_TXD2",
+		"MPP_31_GE_TXD3";
+};
+
+&gpio1 {
+	gpio-line-names =
```

Comment:
Prefer a more compact format for `gpio-line-names` (as many as possible on each line) to match other in-kernel FDTs.